### PR TITLE
New pull request

### PR DESCRIPTION
**Pull request by ModeenF**
In the end this only has the date view change and easy way to add those that have contributed in about view.

### CODE REVIEW

File: src/App.cpp [hunk 1]
```diff
@@ -65,10 +65,15 @@ App::~App()
...
 App::AboutRequested()
 {
+	const char* authors[] = {
+		"Fredrik Modéen.",
+		NULL
+	};
 	BAboutWindow* aboutW = new BAboutWindow(kAppName, kSignature);
 	aboutW->AddDescription(B_TRANSLATE("A native Calendar application for Haiku."));
 	aboutW->AddCopyright(2017, "Akshay Agarwal");
 	aboutW->SetVersion("1.0");
```
Comment (humdingerb) [inline src/App.cpp:75]: I think, if we remove the SetVersion(), the version is picked from the rdef.

File: src/MainWindow.cpp [hunk 2]
```diff
@@ -293,7 +293,7 @@ MainWindow::_InitInterface()
 	fToolBar->AddGlue();
 
 	fSidePanelView = new SidePanelView();
-	fDayView = new DayView(BDate::CurrentDate(B_LOCAL_TIME));
+	fDayView = new DayView(BDate::CurrentDate(B_LOCAL_TIME));
```
Comment (humdingerb) [inline src/MainWindow.cpp:296]: stray tab

File: src/MainWindow.cpp [hunk 3]
```diff
@@ -302,15 +302,27 @@ MainWindow::_InitInterface()
...
-
-	BLayoutBuilder::Group<>(this, B_VERTICAL, 0.0f)
-		.Add(fMenuBar)
-		.Add(fToolBar)
-		.AddGroup(B_HORIZONTAL, 0)
-			.Add(fMainView, 5)
-			.Add(fSidePanelView, 1)
-		.End()
-	.End();
+
```
Comment (humdingerb) [inline src/MainWindow.cpp:305]: As mentioned earlier, I'm still of the opinion that the list-at-right should be the one layout. No options.

File: src/MainWindow.cpp [hunk 4]
```diff
@@ -351,7 +363,7 @@ MainWindow::_SyncWithPreferences()
 		if(preferences->fHeaderVisible == true)
 			fSidePanelView->ShowWeekHeader(true);
 		else
-			fSidePanelView->ShowWeekHeader(false);
+			fSidePanelView->ShowWeekHeader(false);
```
Comment (humdingerb) [inline src/MainWindow.cpp:366]: stray tab

File: src/MainWindow.cpp [hunk 5]
```diff
@@ -378,7 +390,7 @@ MainWindow::_GetSelectedCalendarDate() const
 void
 MainWindow::_ToggleEventViewButton(int selectedButtonId)
 {
-	static const std::vector<int> skEventViewButtonIds = { kDayView, kWeekView,
+/*	static const std::vector<int> skEventViewButtonIds = { kDayView, kWeekView,
```
Comment (humdingerb) [inline src/MainWindow.cpp:393]: Why did you remove the button toggling? That was a welcome change by malbx.

File: src/MainWindow.h [hunk 6]
```diff
@@ -45,7 +44,7 @@ class MainWindow: public BWindow {
 	void			_SyncWithPreferences();
 	void			_UpdateDayView();
 	void			_SetEventListPopUpEnabled(bool state);
-	BDate			_GetSelectedCalendarDate() const;
+	BDate		_GetSelectedCalendarDate() const;
```
Comment (humdingerb) [inline src/MainWindow.h:47]: Accidentally(?) removed tab

File: src/MainWindow.h [hunk 7]
```diff
@@ -68,7 +66,7 @@ class MainWindow: public BWindow {
 	BToolBar*		fToolBar;
 	SidePanelView*		fSidePanelView;
 	DayView*		fDayView;
-	thread_id		fNotificationThread;
+	thread_id		fNotificationThread;
```
Comment (humdingerb) [inline src/MainWindow.h:69]: stray tab

File: src/PreferenceWindow.h [hunk 8]
```diff
@@ -35,14 +35,19 @@ class PreferenceWindow : public BWindow {
 		void				_SyncPreferences(Preferences* preferences);
 		void				_PreferencesModified();
 
-		static const int		kStartOfWeekChangeMessage 		= 1000;
-		static const int 		kShowWeekChangeMessage			= 1001;
-		static const int		kApplyPreferencesMessage		= 1002;
-		static const int		kRevertPreferencesMessage		= 1003;
+		static const int		kStartOfWeekChangeMessage 			= 1000;
```
Comment (humdingerb) [inline src/PreferenceWindow.h:38]: Those tabs look a bit off. You do have them set to 4 spaces per tab?

**Review by humdingerb — COMMENTED** [review]
No need for optional layouts IMO, let's just keep the list-on-right.

File: src/App.cpp (hunk 1 above)
Comment (ModeenF) [reply src/App.cpp:75]: I can remove that..

Comment (ModeenF) [reply src/App.cpp:75]: Yes it did..

File: src/MainWindow.cpp (hunk 2 above)
Comment (ModeenF) [reply src/MainWindow.cpp:296]: Will Fix

File: src/MainWindow.cpp (hunk 3 above)
Comment (ModeenF) [reply src/MainWindow.cpp:305]: But does it hurt to have the option?

File: src/MainWindow.cpp (hunk 4 above)
Comment (ModeenF) [reply src/MainWindow.cpp:366]: Will Fix

File: src/MainWindow.cpp (hunk 5 above)
Comment (ModeenF) [reply src/MainWindow.cpp:393]: sorry leftower as I can't build it in gcc2 with that..

File: src/MainWindow.h (hunk 6 above)
Comment (ModeenF) [reply src/MainWindow.h:47]: No put it 2 tabs from BDate. Same as the other. Tested the above also..

File: src/MainWindow.h (hunk 7 above)
Comment (ModeenF) [reply src/MainWindow.h:69]: Will Fix

File: src/PreferenceWindow.h (hunk 8 above)
Comment (ModeenF) [reply src/PreferenceWindow.h]: Yes I have. But I like to have them in a row. Some how the web don't keep that. But I will se If I can fix that.

**Comment by ModeenF** [comment]
That's new to me.. My changes apers in the pull request.. 

Good that I begun to change the day view in another map, think I need to branch it instead. Looking if BColumnListView can be used in day view. Listing all days time as in Outlook

File: src/MainWindow.cpp (hunk 3 above)
Comment (humdingerb) [reply src/MainWindow.cpp:305]: The Haiku philosophy - which reflects very much my own - is to limit options to the absolute essential stuff. The art is to find the line that separates "essential" from "nice-to-have".
IMO providing one nice layout is OK. People will get used to it, even if the calender app of their other OS looks/works differently. If you start accomondating all tastes and look/workflow of more and more other apps, the deluge of options suffucates the users.
Better nip that in the bud and go with as few options as possible. (Personally, I'd also remove the option for the display of week numbers and just always show those. The start of the month option isn't necessary - use the system locale. Remove the option for the default category -  use the one of the last created task and chances are the default category is the one most used. Presto: no more preferences (for now, until actually needed options show up). )

File: src/MainWindow.h (hunk 6 above)
Comment (humdingerb) [reply src/MainWindow.h:47]: Functions above and below use 3 tabs.

File: src/PreferenceWindow.h (hunk 8 above)
Comment (humdingerb) [reply src/PreferenceWindow.h]: I check in Koder. It's off there too.

File: src/MainWindow.cpp (hunk 5 above)
Comment (humdingerb) [reply src/MainWindow.cpp]: I've moved to 64bit recently. What's the issue exactly?
If it cannot be made to work with gcc2, we'll have to use the 2ndary architecture (gcc8+) on 32bit as well.

File: src/MainWindow.cpp (hunk 3 above)
Comment (ModeenF) [reply src/MainWindow.cpp:305]: It are? Thought that it was default should always work and you can choose things later. Linux way (thats why I don't like Linux) are this are what is and your stuck with it, If you don't like move on :)

File: src/MainWindow.h (hunk 6 above)
Comment (ModeenF) [reply src/MainWindow.h:47]: No not here with tabs set to 4 spaces. not even this sites has them to 3 tabs.

File: src/MainWindow.cpp (hunk 3 above)
Comment (ModeenF) [reply src/MainWindow.cpp:305]: anyhow. Not have to many option can be good but to less option are even worse. And I guess that the creator of the app wanted to have it on the other side he pobably would have put it there :)

File: src/MainWindow.cpp (hunk 5 above)
Comment (ModeenF) [reply src/MainWindow.cpp]: thinking of moving to gcc8 any way..

File: src/MainWindow.cpp (hunk 3 above)
Comment (humdingerb) [reply src/MainWindow.cpp:305]: My experience of many apps on other plattforms is that you often have pages of options that control every little aspect of an app that only a tiny minority of users ever uses. Many options bloat the app over time, maybe in order to justify yet another release.

This being open source, we can discuss and change the GUI to improve it. The original decisions of the first contributer to the app may not be the best or even the product of long contemplation.

File: src/MainWindow.h (hunk 6 above)
Comment (humdingerb) [reply src/MainWindow.h:47]: The checked out code shows 3 tabs between "void			_SetEventListPopUpEnabled(bool state);" for example, and 2 tabs between "BDate		_GetSelectedCalendarDate() const;"

File: src/MainWindow.cpp (hunk 3 above)
Comment (ModeenF) [reply src/MainWindow.cpp:305]: I have thought about it and what other apps do. And I guess your right. Mediaplayer has alot of option but non are about the GUI.. Will remove it. Atelast it's permanently where I llike it ;)

File: src/MainWindow.h (hunk 6 above)
Comment (ModeenF) [reply src/MainWindow.h:47]: But some how it looks like it's 3 tabs so it's futher to the left? What should it be? even tab's or even rows?

**Comment by ModeenF** [comment]
Or it didn't.. pergaps it was some miss reading with WebPositive

**Comment by ModeenF** [comment]
Or it didn't.. pergaps it was some miss reading with WebPositive

File: src/MainWindow.cpp (hunk 3 above)
Comment (humdingerb) [reply src/MainWindow.cpp:305]: +1

File: src/MainWindow.h (hunk 6 above)
Comment (humdingerb) [reply src/MainWindow.h:48]: No matter. Just keep the indentation within the blocks the same. This is how the file looks in Koder:

Koder

File: src/App.cpp [hunk 9]
```diff
@@ -65,10 +65,14 @@ App::~App()
...
 {
+	const char* authors[] = {
+		"Fredrik Modéen.",
+		NULL
+	};
 	BAboutWindow* aboutW = new BAboutWindow(kAppName, kSignature);
 	aboutW->AddDescription(B_TRANSLATE("A native Calendar application for Haiku."));
 	aboutW->AddCopyright(2017, "Akshay Agarwal");
-	aboutW->SetVersion("1.0");
+	aboutW->AddAuthors(authors);
```
Comment (humdingerb) [inline src/App.cpp:75]: stray tab.

File: src/MainWindow.cpp [hunk 10]
```diff
@@ -302,13 +302,14 @@ MainWindow::_InitInterface()
 	BLayoutBuilder::Group<>(fMainView, B_VERTICAL, 0.0f)
 		.Add(fDayView)
 	.End();
-
+
```
Comment (humdingerb) [inline src/MainWindow.cpp:305]: stray tab. sorry about those... I recommend using Koder with the "trailing whitespace highlight" setting on.

File: src/MainWindow.cpp [hunk 11]
```diff
@@ -302,13 +302,14 @@ MainWindow::_InitInterface()
 	BLayoutBuilder::Group<>(fMainView, B_VERTICAL, 0.0f)
 		.Add(fDayView)
 	.End();
-
+	
+	Preferences* preferences = ((App*)be_app)->GetPreferences();
```
Comment (humdingerb) [inline src/MainWindow.cpp:306]: not needed anymore.

File: src/Preferences.h [hunk 12]
```diff
@@ -25,7 +25,7 @@ class Preferences {
 
 	int32					fStartOfWeekOffset;
 	bool					fHeaderVisible;
-	BString					fDefaultCategory;
+	BString				fDefaultCategory;
```
Comment (humdingerb) [inline src/Preferences.h:28]: re-add that removed tab

File: src/MainWindow.h (hunk 6 above)
Comment (ModeenF) [reply src/MainWindow.h:48]: What's that? that's not Pe? I like that Editor better :)

Comment (ModeenF) [reply src/MainWindow.h:48]: Found the aswer below :), Koder.. Will switch to that..

Comment (ModeenF) [reply src/MainWindow.h:48]: Don't know How you get images in here. I have to move to Windows for that.. any how I had a image from Pe and they show them in line.. So we can say that Koder are a must as they show the code differently :)

**Comment by ModeenF** [comment]
lol.. In the end it's more removing stray tabs :).. Whe I'm on it Should I change menu App to Calendar?

**Comment by humdingerb** [comment]
For some reason, I see some of your comments only in the email notification from gitub, not on this page...

> Don't know How you get images in here. I have to move to Windows for that.. any how I had a image from Pe and they show them in line.. So we can say that Koder are a must as they show the code differently :)

I upload images to a site online and link to it with:
! [description] (URL) (without the spaces)

Pe should show the same when used with a monospace font and set to 4 spaces/tab.

> When I'm on it Should I change menu App to Calendar?

If you want to. I still like the menu always being "App", but as seen in recent comments, I may be in the minority...

**Comment by ModeenF** [comment]
I have't check my mail, usualy it's not all there. App or not. To me it's secoindary ;)  I have moved on to the day/week view. Think that BColumnView are the way to go. But I will know more when I have some test views. But If you look at the #67 (I think) and those views of Outlook and Mac (I think) they are more or less nice looking columns, one with times, and one for eatch day. Or am I wring? Perhaps another discussion in another ticket (I'll copy this to another ticket)

**Comment by humdingerb** [comment]
> Think that BColumnView are the way to go. But I will know more when I have some test views. But If you look at the #67 (I think) and those views of Outlook and Mac (I think) they are more or less nice looking columns, one with times, and one for eatch day. Or am I wring? 

You're not wring. :)
The nice thing with Outlook (IIRC, my use of Outlook at a job was a few years ago) is that a task spans the time rows for its duration. I.e. you see how long your appointment is at the height of the 'box' (e.g. from 8:00 to 11:30).
Things become interesting if you have several time overlapping appointments...

Further discussion should be moved to a new issue. This PR looks ready to be merged.
